src/lib.rs

```diff
@@ -60,6 +60,20 @@ pub use fields::TemporalFields;
 /// The `Temporal` result type
 pub type TemporalResult<T> = Result<T, TemporalError>;
 
+/// A library specific trait for unwrapping assertions.
+pub(crate) trait TemporalUnwrap {
+    type Output;
+    fn temporal_unwrap(self) -> TemporalResult<Self::Output>;
+}
+
+impl<T> TemporalUnwrap for Option<T> {
+    type Output = T;
+
+    fn temporal_unwrap(self) -> TemporalResult<Self::Output> {
+        self.ok_or(TemporalError::assert())
+    }
```

Comment:
It would be good to add a `debug_assert`  here so that debug builds panic, and runtime builds throw the error.